### PR TITLE
(backport) feat: add terraform modules for tensorboard controller and web-app charms

### PR DESCRIPTION
**Pull request by NohaIhab**
backports #142 to `track/1.9`
ref #140 and #141

### CODE REVIEW

File: .github/workflows/integrate.yaml [hunk 1]
```diff
@@ -54,6 +54,22 @@ jobs:
...
+  terraform-checks:
+    name: Terraform
+    uses: canonical/charmed-kubeflow-workflows/.github/workflows/terraform-checks.yaml@main
+    strategy:
+      matrix:
+        charm:
+          - tensorboard-controller
+          - tensorboards-web-app
+    with:
+      charm-path: ./charms/${{ matrix.charm }}
```
Comment (mvlassis) [inline .github/workflows/integrate.yaml:66]: Let's add the proper channel here so we don't forget when we actually apply this action
```suggestion
      charm-path: ./charms/${{ matrix.charm }}
      channel: "1.9/stable"
```

Comment (NohaIhab) [reply .github/workflows/integrate.yaml:66]: is this necessary? the default channel of the terraform module in this track is now `1.9/stable` so the action will apply from that channel. I think we can avoid setting this in 2 places.

Comment (mvlassis) [reply .github/workflows/integrate.yaml:66]: From my understanding, [this line](https://github.com/canonical/charmed-kubeflow-workflows/blob/e6d94ca3fee0fc39d85dad66cfeaf19ec5ef9b53/.github/workflows/terraform-apply.yaml#L49) from the action always sets the channel from the workflow's input, and if the input doesn't exist, the default value (so "latest/stable" found [here](https://github.com/canonical/charmed-kubeflow-workflows/blob/e6d94ca3fee0fc39d85dad66cfeaf19ec5ef9b53/.github/workflows/terraform-apply.yaml#L18)) is used. Is that correct?

Comment (NohaIhab) [reply .github/workflows/integrate.yaml:66]: yes that makes sense^ I got in touch with the team and we agreed to have the channel set in the workflow for `track/<version>` branches, this applies to all backport PRs
good catch!

Comment (NohaIhab) [reply .github/workflows/integrate.yaml:66]: as for the default `channel` in the `variables.tf`, this is subject to change based on feedback from the terraform WG

Comment (NohaIhab) [reply .github/workflows/integrate.yaml:66]: the conversation has been resolved, we should use <version>/stable

**Review by mvlassis — APPROVED** [review]
Nice!

**Pull request merged**